src/nupic/algorithms/Cells4.cpp

```diff
@@ -2827,13 +2816,14 @@ void Cells4::computeForwardPropagation(CState& state)
 {
   // Zero out previous values
   // Using memset is quite a bit faster on laptops, but has almost no effect
-  // on Neo15!
+  // on Neo15! // TODO shouldn't we optimize for generic (laptop) CPUs rather than
+  // a cellphone CPU
   _inferActivity.reset();
 
   // Compute cell and segment activity by following forward propagation
   // links from each source cell.  _cellActivity will be set to the total
   // activity coming into a cell.
-#if defined(NTA_ARCH_32) && defined(NTA_OS_DARWIN)
+#ifdef NTA_ARCH_64
```

Comment:
I believe the 32/64 guard was wrong there (as the code looks same as above in https://github.com/breznak/nupic.core/commit/26affc98c0a8c3c61d47c0d14de039119fc44d47#diff-4c232a262b217bd7ab4e516856187881R1328) , the code would still work, but not at optimal speed. This fixes that and sets correct UInt64 for 64bit OS,...